ionoscloud/data_source_dbaas_redisdb_replicaset.go

```diff
@@ -0,0 +1,208 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	redisdb "github.com/ionos-cloud/sdk-go-dbaas-redis"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func dataSourceDBaaSRedisDBReplicaSet() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDBaaSRedisDBReplicaSetRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "The ID of the Redis Replicaset.",
+				Optional:    true,
+			},
+			"location": {
+				Type:             schema.TypeString,
+				Description:      "The cluster location",
+				Optional:         true,
```

Comment:
@avirtopeanu-ionos `location` should be required, on MariaDB it is optional because of the poor design in the beginning, but here it should be required.

Reply:
thanks, I'll add this